File: src/main/java/io/github/hdzitao/editstarters/initializr/InitializrChain.java
package io.github.hdzitao.editstarters.initializr;

import io.github.hdzitao.editstarters.ui.ShowErrorException;

/**
 * Initializr处理链
 *
 * @version 3.2.0
 */
public class InitializrChain {
    private final Initializr[] chain;

    private int i = 0;

    public InitializrChain(Initializr[] chain) {
        this.chain = chain;
    }

    public void initialize(InitializrRequest request, InitializrResponse response) throws Exception {
        if (i >= chain.length) {
            throw new ShowErrorException("Initialization Failed!");
        }

        chain[i++].initialize(request, response, this);
    }
}

File: src/main/kotlin/io/github/hdzitao/editstarters/initializr/InitializrChain.kt
package io.github.hdzitao.editstarters.initializr

import io.github.hdzitao.editstarters.ui.ShowErrorException

/**
 * Initializr处理链
 *
 * @version 3.2.0
 */
class InitializrChain(private val chain: Array<Initializr>) {
    private var i = 0

    fun initialize(request: InitializrRequest, response: InitializrResponse) {
        if (i >= chain.size) {
            throw ShowErrorException("Initialization Failed!")
        }

        chain[i++].initialize(request, response, this)
    }
}
